src/app/portlets/dot-edit-page/dot-edit-page-resolver.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Observable } from 'rxjs';
+import { ActivatedRouteSnapshot, Resolve } from '@angular/router';
+import { PageViewService } from './../../api/services/page-view/page-view.service';
+import { Injectable } from '@angular/core';
+import { PageView } from './shared/models/page-view.model';
+
+@Injectable()
+export class PageViewResolver implements Resolve<any> {
+    constructor(
+        private pageViewService: PageViewService
+    ) {}
+
+    /**
+     * Route resolver for layout/:url
+     * @param {ActivatedRouteSnapshot} route
+     * @returns {Observable<any>}
+     * @memberof PageViewResolver
+     */
+    resolve(route: ActivatedRouteSnapshot): Observable<any> {
+        return this.getPageView(route.queryParams.url);
```

Comment:
Extra private method is unnecessary here, it's ok when you need to call one or the other but in this case is just one, you can go:

```return this.pageViewService.get(route.queryParams.url);```